packages/runtime/src/remote/index.ts

```diff
@@ -0,0 +1,311 @@
+import {
+  type ModuleInfo,
+  type GlobalModuleInfo,
+  isBrowserEnv,
+  warn,
+} from '@module-federation/sdk';
+import { globalLoading } from '../global';
+import {
+  Options,
+  UserOptions,
+  PreloadAssets,
+  PreloadOptions,
+  PreloadRemoteArgs,
+  Remote,
+  RemoteInfo,
+} from '../type';
+import { FederationHost, getRemoteModuleAndOptions } from '../core';
+import {
+  PluginSystem,
+  AsyncHook,
+  AsyncWaterfallHook,
+  SyncHook,
+} from '../utils/hooks';
+import { assert, getRemoteInfo, getRemoteEntryUniqueKey } from '../utils';
+import { DEFAULT_REMOTE_TYPE, DEFAULT_SCOPE } from '../constant';
+import { Module, ModuleOptions } from '../module';
+import { formatPreloadArgs, preloadAssets } from '../utils/preload';
+
+export class RemoteHandler {
```

Comment:
The test cases can be adapted to the new structure

Reply:
Do you mean use RemoteHandler to test registerRemotes or other api ? 

This is clear for developers , but not clear for users . Now users can see out test to know how to use the api and the effect. I think maybe remain current test cases is better, what do you think?